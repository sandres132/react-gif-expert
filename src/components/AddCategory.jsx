import PropTypes from 'prop-types'
import React, { useState } from 'react'

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({target}) => {
        setInputValue(target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault();
        const newCat = inputValue.trim();
        if(newCat.length <= 0) return;
        // setCategories(categories => [inputValue, ...categories ]);
        onNewCategory(newCat);
        setInputValue('');
    }

  return (
    <form onSubmit={onSubmit} aria-label='form'>
        <input 
            type="text" 
            placeholder='Buscar Gifs'
            value={inputValue}
            onChange={onInputChange}
        />
    </form>
  )
}

AddCategory.propTypes ={
    onNewCategory: PropTypes.func.isRequired,
}
