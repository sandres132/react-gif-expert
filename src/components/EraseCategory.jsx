import React from 'react';

import {Eraser} from 'react-bootstrap-icons';

export const EraseCategory = ({cat,onEraseCategory}) => {
    
    // onEraseCategory(cat);
    console.log(cat)

    const onErase = (event) => {
        event.preventDefault();
        const newCat = inputValue.trim();
        if(newCat.length <= 0) return;
        // setCategories(categories => [inputValue, ...categories ]);
        onNewCategory(newCat);
        setInputValue('');
    }

    return (
        <>
            <button key={cat.concat('2')} onClick={onErase}><Eraser color='red' size='15px'/></button>
        </>
    )
}
