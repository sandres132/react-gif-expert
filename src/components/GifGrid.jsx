import React from 'react';

import { GifItem } from "./GifItem";
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({cat}) => {

  const {images, isLoading} = useFetchGifs(cat);
  
  return (
    <>
        <h4>{cat}</h4>
        {
          isLoading && (<h2>Loading...</h2>)
        }
        
        <div className='card-grid'>
          {
            images.map( (img) => (
              <GifItem key={img.id} {...img}/>
            ))
          }
        </div>
        
    </>
    
  )
}

GifGrid.propTypes ={
  cat: PropTypes.string.isRequired,
}
