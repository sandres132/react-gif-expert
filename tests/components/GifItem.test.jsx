import React from 'react';

import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en <GifItem/>', () => {
    const title = 'Luffy GIF - Luffy - Discover & Share GIFs';
    const url = 'https://i.pinimg.com/originals/ec/5f/37/ec5f377a267f9e999a5f4b13bd0fb102.gif';

    test('Debe hacer match con el snapshot', () => {

        const {container} = render(<GifItem title={title} url={url} id={1} />);
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifItem title={title} url={url} id={1} />);
        //screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);

    });

    test('Debe de mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url} id={1} />);
        expect(screen.getByText(title)).toBeTruthy();
    })


})