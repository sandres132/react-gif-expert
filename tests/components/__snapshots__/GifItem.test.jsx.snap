// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Luffy GIF - Luffy - Discover & Share GIFs"
      src="https://i.pinimg.com/originals/ec/5f/37/ec5f377a267f9e999a5f4b13bd0fb102.gif"
    />
    <p>
      Luffy GIF - Luffy - Discover & Share GIFs
    </p>
  </div>
</div>
`;
